### ISSUE TITLE
Milestone 2 -Feedback

### ISSUE THREAD

**Issue by danielramandi** [UBC-MDS/chocolate_exploration#58]
Congratulations on finishing milestone 2! We can see you put a lot of work into this project, nice work! Below we list some specific feedback you can use to improve your project. We provide tick boxes for you to use in the future as you address these concerns to improve the final grade of your project. If anything is unclear, please feel free to ask questions in this issue thread.

**1. Analysis code is abstracted to four (or more) scripts**

- [ ] Command line arguments are missing from one or more scripts -3 accuracy

**Comment:** define output for the train_test_split script.

**1.2 Report is written in a literate code document, and the code in the literate code document primarily functions to source analysis artifacts and format the document.**

- [ ] The proposal.md file has been created and moved to the doc directory -2 mechanics

**Comment:** Refer to the Milestone 2 submission guidelines for details on proposal.md file.

- [ ] The report must contain at least 4 citations relevant to the project. -3 quality

**Comment:** You can add more discussion and intro along with a few citations.

1.3 Code is abstracted to functions with good function design

- [ ] There are cases of functions that do not check for proper input and/or do not provide useful error messages. -1 mechanics

**Comment:** Reproducible scripts should always check for the inputs (type, size, etc.) and provide error messages if an input is missing or is not of correct type.

**1.6 Tests are written for each function, and work expected** - Total 7 points if there are test in all the functions and the tests run

- [ ] There is no documentation. It is not clear what the tests are testing. -3 quality

**Comment:** some functions do not have proper tests, or not documented. You can use unit testing to solve this. [Here](https://towardsdatascience.com/unit-testing-in-r-68ab9cc8d211)’s a nice article on unit testing. But simply, you can check for input type discrepancies and raise proper errors manually.  

**4. Submission expectations**

- [ ] Please provide the link to the release as https://github.com/UBC-MDS/REPOSITORY/tree/VERSION (replacing REPOSITORY and VERSION with the correct ones for your project version for each milestone). This makes is a lot easier for someone to just visit the repository as it was at that point in the project's history. -3 mechanics
